led_strip/src/led_strip_spi_dev.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * SPDX-FileCopyrightText: 2022-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdlib.h>
+#include <string.h>
+#include <sys/cdefs.h>
+#include "esp_log.h"
+#include "esp_check.h"
+#include "esp_rom_gpio.h"
+#include "driver/spi_master.h"
+#include "driver/spi_common.h"
+#include "soc/spi_periph.h"
+#include "led_strip.h"
+#include "led_strip_interface.h"
+#include "hal/spi_hal.h"
+
+
+#define LED_STRIP_SPI_DEFAULT_RESOLUTION (2.5 * 1000 * 1000) // 2.5MHz resolution
+#define LED_STRIP_SPI_DEFAULT_TRANS_QUEUE_SIZE 4
+//Each color of 1 bit is represented by 3 bits, a color occupies 24 bits, 3 byte
```

Comment:
3bits of what?

Reply:
and a color is not always 24bits. If the `led_pixel_format_t` is set to `LED_PIXEL_FORMAT_GRBW`, then it's 32bits for each color